strings.go

```diff
@@ -0,0 +1,368 @@
+		}
+
+		lit, ok := node.(*ast.BasicLit)
+		if !ok {
+			return true
+		}
+
+		if lit.Kind != token.STRING {
+			return true // we only want to obfuscate strings for now
+		}
+
+		value := litToString(lit.Value)
+
+		ciphertext, err := encAes([]byte(value), key)
+		if err != nil {
+			log.Println("Could not encrypt string:", err)
+			return true
+		}
+
+		cursor.Replace(ciphertextStmt(ciphertext))
+
+		return true
+	}
+
+	for _, file := range files {
+		file = astutil.Apply(file, rmConst, obfusStrings).(*ast.File)
+	}
+
+	return files
+}
+
+func litToString(lit string) string {
+	if strings.HasPrefix(lit, "\"") {
+		return strings.Trim(lit, "\"")
+	}
+
+	if strings.HasPrefix(lit, "`") {
+		return strings.Trim(lit, "`")
+	}
+
+	return lit
+}
+
+// ast definitions for injection
+var (
+	aesCipherStmt = &ast.AssignStmt{
+		Lhs: []ast.Expr{
+			&ast.Ident{
+				Name: "block",
+			},
+			&ast.Ident{
+				Name: "_",
+			},
+		},
+		Tok: token.DEFINE,
+		Rhs: []ast.Expr{
+			&ast.CallExpr{
+				Fun: &ast.SelectorExpr{
+					X: &ast.Ident{
+						Name: "aes",
+					},
+					Sel: &ast.Ident{
+						Name: "NewCipher",
+					},
+				},
+				Args: []ast.Expr{
+					&ast.Ident{
+						Name: "garbleKey",
+					},
+				},
+			},
+		},
+	}
+
+	aesGcmCipherStmt = &ast.AssignStmt{
+		Lhs: []ast.Expr{
+			&ast.Ident{
+				Name: "aesgcm",
+			},
+			&ast.Ident{
+				Name: "_",
+			},
+		},
+		Tok: token.DEFINE,
+		Rhs: []ast.Expr{
+			&ast.CallExpr{
+				Fun: &ast.SelectorExpr{
+					X: &ast.Ident{
+						Name: "cipher",
+					},
+					Sel: &ast.Ident{
+						Name: "NewGCM",
+					},
+				},
+				Args: []ast.Expr{
+					&ast.Ident{
+						Name: "block",
+					},
+				},
+			},
+		},
+	}
+
+	plaintextStmt = &ast.AssignStmt{
+		Lhs: []ast.Expr{
+			&ast.Ident{
+				Name: "plaintext",
+			},
+			&ast.Ident{
+				Name: "_",
+			},
+		},
+		Tok: token.DEFINE,
+		Rhs: []ast.Expr{
+			&ast.CallExpr{
+				Fun: &ast.SelectorExpr{
+					X: &ast.Ident{
+						Name: "aesgcm",
+					},
+					Sel: &ast.Ident{
+						Name: "Open",
+					},
+				},
+				Args: []ast.Expr{
+					&ast.Ident{
+						Name: "nil",
+					},
+					&ast.SliceExpr{
+						X: &ast.Ident{
+							Name: "ciphertext",
+						},
+						High: &ast.BasicLit{
+							Kind:  token.INT,
+							Value: "12",
+						},
+						Slice3: false,
+					},
+					&ast.SliceExpr{
+						X: &ast.Ident{
+							Name: "ciphertext",
+						},
+						Low: &ast.BasicLit{
+							Kind:  token.INT,
+							Value: "12",
+						},
+						Slice3: false,
+					},
+					&ast.Ident{
+						Name: "nil",
+					},
+				},
+			},
+		},
+	}
+
+	returnStmt = &ast.ReturnStmt{
+		Results: []ast.Expr{
+			&ast.CallExpr{
+				Fun: &ast.Ident{
+					Name: "string",
+				},
+				Args: []ast.Expr{
+					&ast.Ident{
+						Name: "plaintext",
+					},
+				},
+			},
+		},
+	}
+)
+
+func funcStmt() *ast.FuncDecl {
+	stmts := []ast.Stmt{
+		aesCipherStmt,
+		aesGcmCipherStmt,
+		plaintextStmt,
+		returnStmt,
```

Comment:
why give these statement names, if you're going to just use them as a list? when declaring them globally, just declare them as a list with a single name instead of four.

Reply:
Done